frontend/src/components/pages/RegistrantExperience/WaiverPage/index.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import {
+  Box,
+  Text,
+  Heading,
+  VStack,
+  StackDivider,
+  Stack,
+  Checkbox,
+  ListItem,
+  OrderedList,
+} from "@chakra-ui/react";
+import { ReactJSXElement } from "@emotion/react/types/jsx-namespace";
+import {
+  OptionalClauseResponse,
+  RequiredClauseResponse,
+  WaiverActions,
+  WaiverInterface,
+} from "./waiverTypes";
+
+interface WaiverPageProps {
+  waiverInterface: WaiverInterface;
+  waiverDispatch: any;
+}
+
+const WaiverPage = ({
+  waiverInterface,
+  waiverDispatch,
+}: WaiverPageProps): React.ReactElement => {
+  const optionalClausesJSX: ReactJSXElement[] = [];
+  const requiredClausesJSX: ReactJSXElement[] = [];
+
+  waiverInterface.optionalClauses?.forEach(
+    (clause: OptionalClauseResponse, index: number) => {
+      optionalClausesJSX.push(
+        <Checkbox
+          isChecked={clause.agreed}
+          onChange={() =>
+            waiverDispatch({
+              type: WaiverActions.CLICKOPTIONALCLAUSE,
+              payload: clause,
+            })
+          }
+          key={index}
+        >
+          {clause.text}
+        </Checkbox>,
+      );
+    },
+  );
+  waiverInterface.requiredClauses?.forEach(
+    (clause: RequiredClauseResponse, index: number) => {
+      requiredClausesJSX.push(<ListItem key={index}>{clause.text}</ListItem>);
+    },
+  );
```

Comment:
Yeah, I agree that map would help with readability. I'll change it. However, I'd like to note that the implmentation will look a little different because I'll be using `Radio` instead of `Checkbox`. I just realized that I pushed an old commit (which uses checkbox instead of radio) and I deleted the newer branch on local.

Reply:
How about something like:

https://github.com/uwblueprint/focus-on-nature/blob/d8ab1050df4f6fdb517228f8edaf79a001761a90/frontend/src/components/pages/CampCreation/ScheduleSessions/ScheduledSessionsCard.tsx#L47-L54

So it becomes
```ts
{waiverInterface.requiredClauses?.map(
    (clause: RequiredClauseResponse, index: number) => (<ListItem key={index}>{clause.text}</ListItem>))}
```

Reply:
Ahhh, ok sounds good.